### PR TITLE
Document upgrade guide to 1.0.0

### PR DESCRIPTION
**Pull request by QuentinBisson**
This PR:

* documents the upgrade guide to 1.0.0 https://github.com/giantswarm/roadmap/issues/2879

### Checklist

- [x] Update changelog in CHANGELOG.md.
- [ ] Make sure `values.yaml` and `values.schema.json` are valid. Not needed

### CODE REVIEW

**Review by TheoBrigitte — CHANGES_REQUESTED** [review]

File: docs/upgrade.yaml
Comment (QuentinBisson) [reply docs/upgrade.yaml:26]: Sure thing, I wanted a first eye well first

**Comment by QuentinBisson** [comment]
@TheoBrigitte (as you requested changes) and @marieroque (as you tested the bundle breaking change) could you take a look ? I will not merge until you both approved :)

File: docs/upgrade.md
Comment (marieroque) [inline docs/upgrade.md:25]: I'd mention all possible changes:
- prometheus-agent => prometheusAgent
- promtail-app => promtail
- prometheus-operator-app => kubePrometheusStack
- prometheus-operator-crd => prometheusOperatorCrd

File: docs/upgrade.md
Comment (hervenicol) [inline docs/upgrade.md:19]: This doc talks about upgrading observability bundle to release 1.0.0, and here "upgrade to release 20.0.0" seems out of context.
Don't we have a name for the "Releases"? Like "GiantSwarm Platform  release 20.0.0"?

Comment (QuentinBisson) [reply docs/upgrade.md:19]: Probablye yes

File: docs/upgrade.md
Comment (QuentinBisson) [reply docs/upgrade.md:25]: Yes but also customers are not using it so I did not want to confuse them

Comment (QuentinBisson) [reply docs/upgrade.md:26]: Done

File: docs/upgrade.md
Comment (QuentinBisson) [reply docs/upgrade.md]: DOne thanks

**Review by marieroque — APPROVED** [review]
LGTM

**Comment by QuentinBisson** [comment]
This will merge once @TheoBrigitte approves when he can

**Comment by QuentinBisson** [comment]
Thank you for your reviews :)

File: docs/upgrade.md
Comment (TheoBrigitte) [inline docs/upgrade.md:71]:
```suggestion
      # old values, to remove after upgrade
      prometheus-operator-app:
        namespace: monitoring
      # new values, to add before upgrade
      kubePrometheusStack:
        namespace: monitoring
```

File: docs/upgrade.md
Comment (TheoBrigitte) [inline docs/upgrade.md:89]:
```suggestion
  values: |
    # new values, to add before upgrade
    userConfig:
```

File: docs/upgrade.md
Comment (TheoBrigitte) [inline docs/upgrade.md:98]:
```suggestion
                  existingClaim: grafana
    # old values, to remove after upgrade
    apps:
```

**Review by TheoBrigitte — APPROVED** [review]

**Pull request merged**
